src/vm/threadsuspend.cpp

```diff
@@ -8231,6 +8247,127 @@ void ThreadSuspend::SuspendEE(SUSPEND_REASON reason)
 #endif //TIME_SUSPEND
 }
 
+#if defined(FEATURE_HIJACK) && defined(PLATFORM_UNIX)
+
+// This function is called when a GC is pending. It tries to ensure that the current
+// thread is taken to a GC-safe place as quickly as possible. It does this by doing 
+// one of the following:
+//
+//     - If the thread is in native code or preemptive GC is not disabled, there's
+//       nothing to do, so we return.
+//
+//     - If the thread is in interruptible managed code, we will push a frame that
+//       has information about the context that was interrupted and then switch to
+//       premptive GC mode so that the pending GC can proceed, and then switch back.
+//
+//     - If the thread is in uninterruptible managed code, we will patch the return
+//       address to take the thread to the appropriate stub (based on the return 
+//       type of the method) which will then handle preparing the thread for GC.
+//
+void PALAPI HandleGCSuspensionForInterruptedThread(CONTEXT *interruptedContext)
+{
+    Thread *pThread = GetThread();
+
+    if (pThread->PreemptiveGCDisabled() != TRUE)
+        return;
+
+    PCODE ip = GetIP(interruptedContext);
+
+    if (ExecutionManager::IsManagedCode(ip) != TRUE)
+        return;
+
+    Thread::WorkingOnThreadContextHolder workingOnThreadContext(pThread);
+    if (!workingOnThreadContext.Acquired())
+        return;
+
+    EECodeInfo codeInfo(ip);
+    if (!codeInfo.IsValid())
+        return;
+
+    DWORD addrOffset = ip - codeInfo.GetStartAddress();
```

Comment:
This should be codeInfo.GetRelOffset() instead.

It is not just a cosmetic fix - there is a real bug. NGen does hot-cold code splitting. The method code can be split into to fragments that are far apart. The reloffset returned by codeInfo for the purpose of GCInfo computation is the logical reloffset as if the hot and cold blocks were next to each other.